plugins/inputs/modbus/configuration_request.go

```diff
@@ -144,6 +145,11 @@ func (c *ConfigurationPerRequest) Check() error {
 
 		// Check the fields
 		for fidx, f := range def.Fields {
+			// Name is mandatory
+			if f.Name == "" {
+				return fmt.Errorf("empty field name in request for slave %d", def.SlaveID)
+			}
+
```

Comment:
No! You are changing the logic here. Currently, the name is _only_ mandatory if you do **not** omit the field, i.e. if you specify `omit = true` the field does not need a name. With your change, this is not true anymore.
Anyway, this change is completely unrelated to the topic of your PR. If there is a reason to change this, please submit an additional PR.

Reply:
The problem is that searching for name duplicates requires the name to exist. Also, `omit` does not exist in the configuration_register syntax.

Reply:
Well first of all you are in `configuration_request.go`. Second, it is insane to force users to specify names for stuff that they throw away anyway. And last but not least. The omitted fields are completely ignored, never get checked for duplicate names and are dropped in `groupFieldsToRequests` so their name is _never_ used anywhere. Where do you see an issue with the current code?